examples/ppo/unit_tests.py

```diff
@@ -0,0 +1,87 @@
+
+import jax
+import flax
+from flax import nn
+import numpy as onp
+
+import numpy.testing as onp_testing
+from absl.testing import absltest
+
+#test GAE
+from main import gae_advantages
+class TestGAE(absltest.TestCase):
+  def test_gae_random(self):
+    # create random data, simulating 4 parallel envs and 20 time_steps
+    envs, steps = 10, 100
+    rewards = onp.random.choice([-1., 0., 1.], size=(steps, envs),
+                                p=[0.01, 0.98, 0.01])
+    terminal_masks = onp.ones(shape=(steps, envs), dtype=onp.float64)
+    values = onp.random.random(size=(steps + 1, envs))
+    discount = 0.99
+    gae_param = 0.95
+    adv = gae_advantages(rewards, terminal_masks, values, discount, gae_param)
+    self.assertEqual(adv.shape, (steps, envs))
+    # test the property A_{t} = \delta_t + \gamma*\lambda*A_{t+1}
+    # for each agent separately
+    for e in range(envs):
```

Comment:
In general, I don't like these tests where you basically implement the logic of the code. In my view, tests should be simple hard coded pairs that each test one aspect of the code. Here you don't actually test terminal_masks I believe.

Reply:
Commithttps://github.com/google/flax/pull/470/commits/518a7f61b86e6015d09c3314077b8b75003be205 added a hardcoded test that includes terminal_masks logic as well. The clumsy logic-repeating test has been removed.